proto/api.go

```diff
@@ -79,6 +79,7 @@ type Request interface {
 	gogoproto.Message
 	// Header returns the request header.
 	Header() *RequestHeader
+	KVHeader() *KVRequestHeader
```

Comment:
Hmm. That's kind of ugly. I don't have a better solution to offer, though. You should add a comment explaining this, otherwise it is surprising to the reader that `Request` has a `KVHeader()` method.